### PR TITLE
feat!: support tree-shaking

### PR DESCRIPTION
**Pull request by Shinigami92**
BREAKING CHANGE: importing of faker changes

### CODE REVIEW

**Comment by Shinigami92** [comment]
when we merge this, we need to write a migration guide or at least provide a RegExp replace pattern for out users

```ts
// old-js
const faker = require('faker')

// to
const { faker } = require('@faker-js/faker')
// or
const faker = require('@faker-js/faker').faker

// --------------------------------------------

// new-js
import faker from 'faker'
// or
import * as faker from 'faker'

// to
import { faker } from '@faker-js/faker'
```

File: examples/node/generateMultiLevelMultiLocaleJSON.js [hunk 1]
```diff
@@ -1,31 +1,39 @@
...
-  console.log(JSON.stringify(company))
-  fs.writeFile(__dirname + '/companyDataSet.json',  JSON.stringify(company), function() {
-    console.log("dataSet generated successfully!");
-  });
\ No newline at end of file
+// This example shows the generation of a multilevel object and JSON document using various faker.js features
+// including name, address, company, date and commerce namespaces, moustache expressions and random element production
+// Using the helper function arr, randomly sized collections of elements are produced in the document.
+
+var faker = require('../../lib').faker;
```
Comment (Shinigami92) [inline examples/node/generateMultiLevelMultiLocaleJSON.js:5]: This is the important line in this file.
I didn't touched the rest of this file, just auto-prettier on save.

File: examples/node/generateSet.js [hunk 2]
```diff
@@ -1,19 +1,26 @@
 var fs = require('fs');
 
-var faker = require('../../index');
-
+var faker = require('../../lib').faker;
```
Comment (Shinigami92) [inline examples/node/generateSet.js:3]: This is the important line in this file.
I didn't touched the rest of this file, just auto-prettier on save.

File: package.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
     "url": "https://github.com/faker-js/faker.git"
   },
   "license": "MIT",
-  "main": "index.js",
+  "main": "lib/index.js",
```
Comment (Shinigami92) [inline package.json:25]: This is the most important line of this PR :slightly_smiling_face:

**Comment by Shinigami92** [comment]
This could affect how the browser version is used :thinking: 
I'm not sure but do we need something to do like this in `src/index.ts`?:
```ts
if (typeof window === 'object') {
  window.faker = faker;
}
```

**Comment by damienwebdev** [comment]
We need an integration test running in CI to validate both Node behavior and browser behavior of the legacy scenario before I think we can merge this. I also think that we move this to `v7` to give our dependents some space to focus on their work rather than a breaking change.

**Comment by Shinigami92** [comment]
> We need an integration test running in CI to validate both Node behavior and browser behavior of the legacy scenario before I think we can merge this. I also think that we move this to `v7` to give our dependents some space to focus on their work rather than a breaking change.

To me it's not such a critical change + I would like to do it now, cause the folks that migrate to us need to rewrite the import statements anyways.
So instead of doing it twice, they just need it to do it once.

**Review by clarkerican — DISMISSED** [review]
Looks good to me!

**Comment by import-brain** [comment]
> We need an integration test running in CI to validate both Node behavior and browser behavior of the legacy scenario before I think we can merge this. I also think that we move this to `v7` to give our dependents some space to focus on their work rather than a breaking change.

By semantic versioning, this would be a "next major version" change anyway, because it would be a possible breaking change.

**Comment by import-brain** [comment]
I'll mark this as ```needs tests``` because we should probably write tests for this to make sure it works before we merge it.

**Review by MilosPaunovic — DISMISSED** [review]

**Comment by Shinigami92** [comment]
> > We need an integration test running in CI to validate both Node behavior and browser behavior of the legacy scenario before I think we can merge this. I also think that we move this to `v7` to give our dependents some space to focus on their work rather than a breaking change.
> 
> By semantic versioning, this would be a "next major version" change anyway, because it would be a possible breaking change.

v6 is currently the next major change, we just released alphas, so nothing wrong with that from my POV

> I'll mark this as `needs tests` because we should probably write tests for this to make sure it works before we merge it.

Could you explain what you mean what test do we need? Currently all ~20k are passing and I'm very happy that we have so much.
This PR does not affect the runtime behavior at all, just improve how you import it.

I may provide a regex-pattern and/or a sed-command so folks can easily update.
But IMO we already do need these patterns right now, cause the import already changed from `faker` to `@faker-js/faker`.

File: src/index.ts [hunk 4]
```diff
@@ -271,5 +271,7 @@ export class Faker {
   }
 }
 
-export default Faker;
-module.exports = Faker;
+// since we are requiring the top level of faker, load all locales by default
+export const faker: Faker = new Faker({
+  locales: require('./locales'),
+});
```
Comment (Shinigami92) [inline src/index.ts:277]: We need to add `export default faker;` here

**Review by JessicaSachs — DISMISSED** [review]
After the export default line is in, this lgtm

**Review by MilosPaunovic — APPROVED** [review]

**Review by JessicaSachs — APPROVED** [review]

**Review by prisis — APPROVED** [review]

**Pull request merged**